src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,901 @@
+            CreateTestFile(TestBase, DestData, NewFileNameWithPath)
+            Assert.Throws(Of IO.IOException)(Sub() FileSystem.CopyDirectory(FullPathToSourceDirectory, FullPathToTargetDirectory, overwrite:=False))
+            Assert.Equal(IO.Directory.GetFiles(FullPathToTargetDirectory).Count, IO.Directory.GetFiles(FullPathToSourceDirectory).Count)
+            For Each CurrentFile As String In IO.Directory.GetFiles(FullPathToTargetDirectory)
+                If CurrentFile.EndsWith("0") Then
+                    ' Make sure file 0 is unchanged with DestDate
+                    Using stream As New IO.StreamReader(CurrentFile)
+                        Dim ReadData(DestData.Length - 1) As Char
+                        stream.Read(ReadData, 0, DestData.Length)
+                        Assert.Equal(DestData, ReadData)
+                    End Using
+                Else
+                    ' Ensure file 1 - 5 transferred SourData
+                    Using stream As New IO.StreamReader(IO.File.OpenRead(CurrentFile))
+                        Dim ReadData(SourceData.Length - 1) As Char
+                        stream.Read(ReadData, 0, SourceData.Length)
+                        Assert.Equal(SourceData, ReadData)
+                    End Using
+                End If
+            Next
+            TestBase.Dispose()
+        End Sub
+
+        <Fact>
+        Public Sub CopyDirectory_SourceDirectoryName_DestinationDirectoryName_OverwriteTrue()
+            Dim TestBase As New FileIOTestBase
+            Dim FullPathToSourceDirectory As String = IO.Path.Combine(TestBase.TestDirectory(), "SourceDirectory")
+            Dim FullPathToTargetDirectory As String = IO.Path.Combine(TestBase.TestDirectory(), "TargetDirectory")
+            IO.Directory.CreateDirectory(FullPathToSourceDirectory)
+            IO.Directory.CreateDirectory(FullPathToTargetDirectory)
+            For i As Integer = 0 To 5
+                CreateTestFile(TestBase, SourceData, IO.Path.Combine(FullPathToSourceDirectory, $"NewFile{i}"))
+            Next
+            FileSystem.CopyDirectory(FullPathToSourceDirectory, FullPathToTargetDirectory, overwrite:=True)
+            Assert.Equal(IO.Directory.GetFiles(FullPathToSourceDirectory).Count, IO.Directory.GetFiles(FullPathToTargetDirectory).Count)
+            For Each CurrentFile As String In IO.Directory.GetFiles(FullPathToTargetDirectory)
+                ' Ensure copy transferred written data
+                Using stream As New IO.StreamReader(IO.File.OpenRead(CurrentFile))
+                    Dim ReadData(SourceData.Length - 1) As Char
+                    stream.Read(ReadData, 0, SourceData.Length)
+                    Assert.Equal(SourceData, ReadData)
+                End Using
+            Next
+            TestBase.Dispose()
+        End Sub
+
+        <Fact(Skip:="Not Implemented")>
+        Public Sub CopyDirectory_SourceDirectoryName_DestinationDirectoryName_UIOption()
```

Comment:
remove this test

Reply:
I thought Dan asked me to leave them in and mark with Not Implemented so I we ever get a UI it will be clear where we need tests. No issue removing them. Ditto for all of them

Reply:
He meant that to throw ```platformNotSupportedException``` in the implementation in the src.